script.js

```diff
@@ -23,4 +21,69 @@ tags.forEach((tag) => {
   tag.addEventListener('click', () => {
     closeMenu();
   });
-});
\ No newline at end of file
+});
+
+const nameError = document.querySelector('#name-error');
+const emailError = document.querySelector('#email-error');
+const submitError = document.querySelector('#submit-error');
+function validate() {
+  const name = document.getElementById('user-name').value;
+  if (name.lenght === 0) {
+    nameError.innerHTML = 'Name is required';
+    return false;
+  }
+  if (!name.match(/^[A-Za-z]*\s{1}[A-Za-z]*$/)) {
+    nameError.innerHTML = 'Invalid name';
+    return false;
+  }
+  nameError.innerHTML = '<img src="checkbox-checked-solid-24 (1).png" alt="check icon">';
+  return true;
+}
+
+function validateEmail() {
+  const email = document.getElementById('user-email').value;
+
+  if (email.lenght === 0) {
+    emailError.innerHTML = 'Email is required';
+    return false;
+  }
+  if (!email.match(/^([a-za-z0-9_\-.])+@([a-za-z0-9_\-.])+\.([a-za-z]{2,4})$/)) {
+    emailError.innerHTML = 'Invalid email';
+    return false;
+  }
+  emailError.innerHTML = '<img src="checkbox-checked-solid-24 (1).png" alt="check icon">';
+  return true;
+}
+
+// LOCAL STORAGE
+const formName = document.getElementById('user-name');
+const email = document.getElementById('user-email');
+const formText = document.getElementById('user-text');
+
+const localData = () => {
+  const userData = {
+    userName: formName.value,
+    UserMail: email.value,
+    txt: formText.value,
+  };
+  localStorage.setItem('userData', JSON.stringify(userData));
+};
+const storedDetails = () => {
+  const info = JSON.parse(localStorage.getItem('userData'));
+  formName.value = info.formName;
+  email.value = info.email;
+  formText.value = info.formText;
+};
+
+function validateForm() {
+  if (!validate() || !validateEmail()) {
+    submitError.style.display = 'block';
+    submitError.innerHTML = 'Please fix errors to submit';
+    setTimeout(() => { submitError.style.display = 'none'; }, 3000);
+    return false;
+  }
+  localData();
+  storedDetails();
+  return true;
+}
```

Comment:
- [ ] I believe your implementation can be more easy to follow and smooth. see below my suggestion, you use the `change` event on each input to execute the `saveFormData()`. This ensures that user-typed data is saved when the input element's value changes. 

**_The code below is an example of how this can be implemented._** 

```javascript
// Function to save form data to local storage
function saveFormData() {
  const nameInput = document.getElementById('name');
  const emailInput = document.getElementById('email');
  const messageInput = document.getElementById('message');

  // Get the form values
  const formData = {
    name: nameInput.value,
    email: emailInput.value,
    message: messageInput.value,
  };

  // Save form data to local storage
  localStorage.setItem('formData', JSON.stringify(formData));
}

// Function to load form data from local storage
function loadFormData() {
  const nameInput = document.getElementById('name');
  const emailInput = document.getElementById('email');
  const messageInput = document.getElementById('message');

  // Check if form data exists in local storage
  if (localStorage.getItem('formData')) {
    const formData = JSON.parse(localStorage.getItem('formData'));

    // Populate the form inputs with the stored values
    nameInput.value = formData.name;
    emailInput.value = formData.email;
    messageInput.value = formData.message;
  }
}

// Event listeners for input changes
document.getElementById('name').addEventListener('change', saveFormData);
document.getElementById('email').addEventListener('change', saveFormData);
document.getElementById('message').addEventListener('change', saveFormData);

// Event listener for form submission
document.getElementById('contactForm').addEventListener('submit', function(e) {
  e.preventDefault(); // Prevent form submission

  // Save the form data to local storage
  saveFormData();

  // Submit the form (optional)
  // this.submit();
});

// Load form data when the page finishes loading
window.onload = loadFormData;
```